pkg/stores/partition/store.go

```diff
@@ -175,17 +300,79 @@ func (s *Store) Watch(apiOp *types.APIRequest, schema *types.APISchema, wr types
 	return response, nil
 }
 
-// getLimit extracts the limit parameter from the request or sets a default of 100000.
-// Since a default is always set, this implies that clients must always be
-// aware that the list may be incomplete.
-func getLimit(req *http.Request) int {
-	limitString := req.URL.Query().Get("limit")
-	limit, err := strconv.Atoi(limitString)
+func toAPI(schema *types.APISchema, obj runtime.Object) types.APIObject {
+	if obj == nil || reflect.ValueOf(obj).IsNil() {
```

Comment:
> I think between these two cases, it is better to assume that the runtime.Object's underlying value is a non-nil struct, than it is to assume that the runtime.Object's underlying value is a struct with a pointer receiver mainly because the latter is not idiomatic behavior in go.

runtime.Object should basically always contain a struct with methods defined for pointer receivers because the types that implement [runtime.Object](https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime#Object) include [metav1.TypeMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#TypeMeta.GetObjectKind) as an [embedded field](https://github.com/rancher/rancher/blob/38979f8ebe5c1ecb0121494d185758741fcb0d5b/pkg/apis/management.cattle.io/v3/public_authn_types.go#L12) and have autogenerated [DeepCopyObject pointer methods](https://github.com/rancher/rancher/blob/38979f8ebe5c1ecb0121494d185758741fcb0d5b/pkg/apis/management.cattle.io/v3/zz_generated_deepcopy.go#L539). So I think we still need reflect no matter what. We could add safety by checking `.Kind` before checking `.IsNil()`.